### ISSUE TITLE
Proper way of versioning Harbor

### ISSUE THREAD

**Issue by casperfrx** [goharbor/harbor#8002]
I couldn't find this question or an answer that relates to it but here's my question:

How to properly version changes to the configuration of Harbor? I suppose only harbor.yml can be versioned but I would like to make some persistent changes to the compose file (adding an external network to a harbor service for example). When using the ./prepare.sh script, these changes are overwritten and there are certificates generated in /common/config.

Basically, how should I version Harbor if I want to make some custom changes which are not available in harbor.yml?

**Comment by bitsf** [comment]
I think this is difficult to implement in Harbor, maybe you can create a patch file, each time after you use prepare script, then you can patch the docker-compose.yml manually, then restart using docker-compose

**Comment by reasonerjt** [comment]
@caniszczyk there's no good way unless you build your own `prepare` image.
